### PR TITLE
SWS-195: Display waiting page until service graph details are returned.

### PR DESCRIPTION
**Pull request by mwringe**
Simple fix if we are waiting for the backend to return the service graph details.

[perhaps a very naive approach, suggestions for a better approach appreciated]

### CODE REVIEW

**Comment by mwringe** [comment]
@mtho11 thoughts?

**Comment by mtho11** [comment]
@mwringe what about adding this PF React [Empty State component](https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html?selectedKind=EmptyState&selectedStory=EmptyState&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

**Comment by mwringe** [comment]
This issue isn't about the empty state. We need to handle the empty state and we should add the empty state component, but that is another task.

This issue is more about what to display when we are waiting for the network to return or an error condition has occurred (both use cases that the empty state is not meant for). Specifically to resolve SWS-195 where we render something incorrect if we have an error or waiting for the network response.

**Comment by mtho11** [comment]
Ah ok, I guess I was conflating the issue.

File: src/components/CytoscapeLayout/CytoscapeLayout.tsx [hunk 1]
```diff
@@ -43,9 +44,24 @@ export default class CytoscapeLayout extends React.Component<CytoscapeLayoutProp
         const elements: { [key: string]: any } = response['data'];
         console.log(elements);
         this.setState(elements);
+
+        let canvasWrapper = document.getElementById('cytoscape-container')!;
```
Comment (mtho11) [inline src/components/CytoscapeLayout/CytoscapeLayout.tsx]: We probably need to eliminate these `getElementById` altogether as they stop reusabilty. We can't reuse these components in any other pages then.
I know there was already one instance of this introduced by @cfcosta.

Comment (vnugent) [reply src/components/CytoscapeLayout/CytoscapeLayout.tsx]: That explains why the test smoke test blew up.  We should re-enabled the [test](https://github.com/swift-sunshine/swsui/pull/75) when addressing `document.getElementById()` 
 
https://travis-ci.org/swift-sunshine/swsui/builds/347950574
```
src/pages/ServiceDetails/ServiceMetrics.tsxsrc/pages/ServiceDetails/ServiceMetrics.tsx 137ms
src/pages/ServiceGraph/ServiceGraphPage.tsxsrc/pages/ServiceGraph/ServiceGraphPage.tsx 15ms
src/pages/ServiceList/ServiceListComponent.tsxsrc/pages/ServiceList/ServiceListComponent.tsx 136ms
src/pages/ServiceList/ServiceListPage.tsxsrc/pages/ServiceList/ServiceListPage.tsx 33ms
src/registerServiceWorker.jssrc/registerServiceWorker.js 45ms
src/registerServiceWorker.tssrc/registerServiceWorker.ts 55ms
src/services/Api.tssrc/services/Api.ts 34ms
src/setupTests.tssrc/setupTests.ts 4ms
src/types/Metrics.tssrc/types/Metrics.ts 4ms
src/types/Namespace.tssrc/types/Namespace.ts 2ms
src/types/NamespaceId.tssrc/types/NamespaceId.ts 2ms
src/types/ServiceFilter.tssrc/types/ServiceFilter.ts 10ms
src/types/ServiceId.tssrc/types/ServiceId.ts 3ms
src/types/ServiceInfo.tssrc/types/ServiceInfo.ts 20ms
src/types/ServiceListComponent.tssrc/types/ServiceListComponent.ts 6ms
/home/travis/build/swift-sunshine/swsui/node_modules/react-scripts-ts/scripts/test.js:20
  throw err;
  ^
TypeError: Cannot read property 'getElementById' of undefined
    at /home/travis/build/swift-sunshine/swsui/src/components/CytoscapeLayout/CytoscapeLayout.tsx:53:95
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
``

File: src/components/CytoscapeLayout/CytoscapeLayout.tsx [hunk 2]
```diff
@@ -43,9 +44,24 @@ export default class CytoscapeLayout extends React.Component<CytoscapeLayoutProp
...
+        if (messageWrapper != null) {
+          messageWrapper.style.display = 'none';
+        }
       })
       .catch(error => {
         this.setState({});
+
+        let messageWrapper = document.getElementById('empty-graph')!;
+        if (messageWrapper != null) {
+          messageWrapper.innerHTML = 'Error Fetching Service Graph';
```
Comment (mtho11) [inline src/components/CytoscapeLayout/CytoscapeLayout.tsx]: We should also send a ToastNotification of this error.

**Comment by mtho11** [comment]
@mwringe this needs rebasing

**Comment by mwringe** [comment]
Its needs more than just rebasing. It need some additional work as well to make it function better as a react component and be able to pass the tests
I am marking this as 'Do Not Merge' until the PR is properly updated

**Comment by mwringe** [comment]
Closing. See https://github.com/swift-sunshine/swsui/pull/98 instead.

**Pull request closed without merging**